validations/src/validations.rs

```diff
@@ -1314,8 +1314,10 @@ pub fn update_utxo_diff(
 }
 
 /// Function to validate transactions in a block and update a utxo_set and a `TransactionsPool`
+///
+/// This uses a `Visitor` that will visit each transaction as well as its fee and weight.
 #[allow(clippy::too_many_arguments)]
-pub fn validate_block_transactions(
+pub fn validate_block_transactions<T>(
```

Comment:
It's a bit strange to have this generic parameter here, I believe it can be avoided by removing the `State` type from the `Visitor` trait. Because a function that takes a visitor does not need to know the internal state of the visitor, and users that create a visitor can also create a dedicated method to get the state.

Reply:
I see. Most funnily, the argument is inferred by the compiler always that an actual visitor is used. Only in the case that `None` is passed instead, we need to specify the type parameter here.

The `<T as Visitor::State>` associated type however is very useful when defining stuff like the `DualVisitor` or the `GenericVisitor` :crying_cat_face:

Reply:
There should be some way to solve this, because this function does never use `T` so it is not actually generic, but not sure how to tell that to the compiler.